src/index.tsx

```diff
@@ -16,11 +18,23 @@ export const withFormik = makeDecorator({
     const channel = addons.getChannel();
     let submitter: () => void;
     channel.on(EVT_SUBMIT, () => submitter && submitter());
-    const formikConfig = parameters as ConfigWithoutSubmit | undefined;
-    const initialValues = (formikConfig && formikConfig.initialValues) || {};
+    const formikConfig = parameters as ConfigWithoutExtra | undefined;
+    let initialValues =
+      context.args || (formikConfig && formikConfig.initialValues) || {};
+
+    if (
+      context.args &&
+      formikConfig &&
+      !formikConfig.initialValues &&
+      formikConfig.validationSchema &&
+      parameters.castArgs
+    ) {
+      initialValues = formikConfig.validationSchema.cast(initialValues);
+    }
 
     return (
       <Formik
+      enableReinitialize
```

Comment:
```suggestion
        enableReinitialize
```

Sorry to nitpick but could you indent properly? (really I should add prettier to the project)